connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SinkConnectorConfig.java

```diff
@@ -57,11 +57,18 @@
     public static final short DLQ_TOPIC_REPLICATION_FACTOR_CONFIG_DEFAULT = 3;
     private static final String DLQ_TOPIC_REPLICATION_FACTOR_CONFIG_DISPLAY = "Dead Letter Queue Topic Replication Factor";
 
+    public static final String DLQ_CONTEXT_HEADERS_ENABLE_CONFIG = DLQ_PREFIX + "context.headers.enable";
+    public static final boolean DLQ_CONTEXT_HEADERS_ENABLE_DEFAULT = false;
+    public static final String DLQ_CONTEXT_HEADERS_ENABLE_DOC = "If true, add headers containing error context to the messages " +
+            "written to the dead letter queue. All error context headers will be prefixed with \"__connect.errors\".";
```

Comment:
Nit:

    To avoid clashing with headers from the original record, all error context header keys 
    will start with <code>__connect.errors.</code>.